packages/aws-cdk/lib/api/hotswap-deployments.ts

```diff
@@ -59,13 +66,58 @@ export async function tryHotswapDeployment(
 
 async function findAllHotswappableChanges(
   stackChanges: cfn_diff.TemplateDiff, evaluateCfnTemplate: EvaluateCloudFormationTemplate,
+  sdk: ISDK, rootStackArtifact: cxapi.CloudFormationStackArtifact, nestedStackNames: {[key: string]: any}, hotswappableResources?: HotswapOperation[],
 ): Promise<HotswapOperation[] | undefined> {
   const resourceDifferences = getStackResourceDifferences(stackChanges);
 
   let foundNonHotswappableChange = false;
   const promises: Array<Array<Promise<ChangeHotswapResult>>> = [];
+
+  if (!hotswappableResources) {
+    hotswappableResources = new Array<HotswapOperation>();
+  }
+
+  // process any resources in nested stacks
+  // if any of them are not hotswappable, fail now
+  const resourceDifferenceEntries = Object.entries(resourceDifferences);
+  for (const [logicalId, change] of resourceDifferenceEntries.filter(
+    ([_, resourceDifference]) => (resourceDifference.newValue?.Type === 'AWS::CloudFormation::Stack' && resourceDifference.oldValue?.Type === 'AWS::CloudFormation::Stack'))
+  ) {
+    const parameters = change.newValue?.Properties?.Parameters;
+    if (parameters) {
+      for (const paramName in parameters) {
+        if (typeof parameters[paramName] !== 'string') {
+          parameters[paramName] = await evaluateCfnTemplate.evaluateCfnExpression(parameters[paramName]);
+
+        }
+      }
+    }
+
+    const nestedStackName = nestedStackNames[logicalId].stackName;
+    const evaluateNestedCfnTemplate = nestedStackName
+      ? evaluateCfnTemplate.createNestedEvaluateCloudFormationTemplate(nestedStackName,
+        sdk, rootStackArtifact, parameters)
+      : evaluateCfnTemplate;
+
+    const nestedDiff = cfn_diff.diffTemplate(change.oldValue?.Properties?.NestedTemplate, change.newValue?.Properties?.NestedTemplate);
+    if (await findAllHotswappableChanges(nestedDiff, evaluateNestedCfnTemplate, sdk,
+      rootStackArtifact, nestedStackNames[logicalId].children, hotswappableResources) === undefined
+    ) {
+      return undefined;
+    }
+  }
+
+  // if a stack has been deleted, fail now
+  for (const [_, change] of resourceDifferenceEntries) {
+    if (change.newValue?.Type !== 'AWS::CloudFormation::Stack' && change.oldValue?.Type === 'AWS::CloudFormation::Stack') {
+      return undefined;
+    }
+  }
+
   // gather the results of the detector functions
-  for (const [logicalId, change] of Object.entries(resourceDifferences)) {
+  for (const [logicalId, change] of resourceDifferenceEntries.filter(
+    ([_, resourceDifference]) => (resourceDifference.newValue?.Type !== 'AWS::CloudFormation::Stack' || resourceDifference.oldValue?.Type !== 'AWS::CloudFormation::Stack'))
+  ) {
     const resourceHotswapEvaluation = isCandidateForHotswapping(change);
```

Comment:
Can you explain why are we iterating through the changes twice, first for the Nested Stacks, then for the non-Nested Stack resources? Why not just use the existing loop, but add some logic to it dealing with Nested Stacks?